src/rdf4cpp/rdf/Quad.cpp

```diff
@@ -0,0 +1,11 @@
+#include "Quad.hpp"
+
+namespace rdf4cpp::rdf {
+Quad::Quad(Node graph, Node subject, Node predicate, Node object) : QuadPattern(graph, subject, predicate, object) {}
```

Comment:
maybe validate on creation as well? :thinking:

Reply:
I would suggest to add an 5th parameter `validate = true`. That validate by default but would leave the user the freedom to first create a Quad and then fill it e.g. in a loop with valid values.

Reply:
fine with me, would there be a use case where it would be good to have it first and validate it later? :thinking:  
I mean normally I would use interfaces like `Quad::Quad(IRINode graph, SubjectNode subject, IRINode predicate, Node object)` or similar, but as this seems to be messy in C++ i think that will work as well. 

Same here with the benefit of using a factory.

Reply:
Ah, or better: we have already a default constructor which initialized Quad with null Terms. So we can validate this constructor always.

Reply:
or, like you commented below: we have a static factory method which does the validation and returns an optional. So we would need to explicitly check if it was valid. The constructor could stay non-validating then.

Reply:
sounds good to me, so if one really knows what they are doing, they can use the constructor directly (e.g. if they know it is definitely valid) 
otherwise the factory can be used.

Reply:
please re-review the updated class.

Reply:
seems good

(we may want to use a rdf4cpp::sec::Result in the future to let the user know what was wrong, but that is debatable and future work)